### PR TITLE
add hypershift upgrade job to CNO presubmits

### PR DESCRIPTION
**Pull request by jluhrsen**
optional:true
always_run:false

### CODE REVIEW

**Comment by flavio-fernandes** [comment]
Nice!!!!

File: ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-release-4.14__4.14-upgrade-from-stable-4.13.yaml [hunk 1]
```diff
@@ -29,6 +29,12 @@ releases:
       include_built_images: true
       name: "4.14"
       namespace: ocp
+  target:
```
Comment (LiangquanLi930) [inline ci-operator/config/openshift/cluster-network-operator/openshift-cluster-network-operator-release-4.14__4.14-upgrade-from-stable-4.13.yaml:32]: you need enable test image in this file
```
  tests-private-preupg:
    name: tests-private
    namespace: ci
    tag: "4.14"
```

**Comment by flavio-fernandes** [comment]
@jluhrsen a humble ask: can you please annotate it in this PR what command(s) I wouls use in order to run this job against a PR in, let's say.... CNO ? :)
I can't wait to try it on  https://github.com/openshift/cluster-network-operator/pull/2063  <3
/cc @jechen0648

File: ci-operator/jobs/openshift/cluster-network-operator/openshift-cluster-network-operator-release-4.14-presubmits.yaml [hunk 2]
```diff
@@ -1,5 +1,86 @@
...
+    decorate: true
+    labels:
+      ci-operator.openshift.io/cloud: aws
+      ci-operator.openshift.io/cloud-cluster-profile: aws-qe
+      ci-operator.openshift.io/variant: 4.14-upgrade-from-stable-4.13
+      ci.openshift.io/generator: prowgen
+      pj-rehearse.openshift.io/can-be-rehearsed: "true"
+    name: pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift-replace-f7
+    optional: true
+    rerun_command: /test 4.14-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift-replace-f7
```
Comment (jluhrsen) [inline ci-operator/jobs/openshift/cluster-network-operator/openshift-cluster-network-operator-release-4.14-presubmits.yaml:19]: @flavio-fernandes , look here 👀

**Comment by jluhrsen** [comment]
> @jluhrsen a humble ask: can you please annotate it in this PR what command(s) I wouls use in order to run this job against a PR in, let's say.... CNO ? :) I can't wait to try it on [openshift/cluster-network-operator#2063](https://github.com/openshift/cluster-network-operator/pull/2063) <3 /cc @jechen0648

I showed you where you can find it in the code, but it's

`/test 4.14-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift-replace-f7`

still, instead of remembering it just type this in your CNO PR comments:

`/test i-dont-know-wth-the-name-of-the-job-is-i-want-to-run
`
and then it will yell at you and tell you all the names of all the jobs and you can just copy/paste :)

**Comment by ricky-rav** [comment]
Would it make sense to drop the final part of the job name and "just" have `pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift`? We understood on slack that `f7` means that the corresponding periodic job is run every 7 days. As for `replace`, no idea, but I guess we don't need it, do we? :)

**Comment by jluhrsen** [comment]
> Would it make sense to drop the final part of the job name and "just" have `pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift`? We understood on slack that `f7` means that the corresponding periodic job is run every 7 days. As for `replace`, no idea, but I guess we don't need it, do we? :)

if we want to keep it with the normal conventions, we'd remove the ipi too.

But, @flavio-fernandes mentioned this jobs is doing multiple things. maybe there is more than just upgrade
that we'd want to indicate with some string appended to the end of the job name. if so, LMK what that is, otherwise
I vote for `pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ovn-hypershift`

**Comment by flavio-fernandes** [comment]
> > Would it make sense to drop the final part of the job name and "just" have `pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift`? We understood on slack that `f7` means that the corresponding periodic job is run every 7 days. As for `replace`, no idea, but I guess we don't need it, do we? :)
> 
> if we want to keep it with the normal conventions, we'd remove the ipi too.
> 
> But, @flavio-fernandes mentioned this jobs is doing multiple things. maybe there is more than just upgrade that we'd want to indicate with some string appended to the end of the job name. if so, LMK what that is, otherwise I vote for `pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ovn-hypershift`

Sorry for my bad wording, @jluhrsen . Yes, all we want is `pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ovn-hypershift`

**Comment by trozet** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift-replace-f7

**Comment by trozet** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift-replace-f7

**Comment by ricky-rav** [comment]
/pj-rehearse pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift-replace-f7

**Comment by jluhrsen** [comment]
> > > Would it make sense to drop the final part of the job name and "just" have `pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ipi-ovn-hypershift`? We understood on slack that `f7` means that the corresponding periodic job is run every 7 days. As for `replace`, no idea, but I guess we don't need it, do we? :)
> > 
> > 
> > if we want to keep it with the normal conventions, we'd remove the ipi too.
> > But, @flavio-fernandes mentioned this jobs is doing multiple things. maybe there is more than just upgrade that we'd want to indicate with some string appended to the end of the job name. if so, LMK what that is, otherwise I vote for `pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ovn-hypershift`
> 
> Sorry for my bad wording, @jluhrsen . Yes, all we want is `pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-ovn-hypershift`

@flavio-fernandes @ricky-rav , you guys ok with  `pull-ci-openshift-cluster-network-operator-release-4.14-4.14-upgrade-from-stable-4.13-aws-hypershift-ovn-upgrade`? using `ovn-upgrade` at the end follows our other job names and hypershift is platform-ish right? so aws-hypershift indicates the "platform" which we like to keep before the CNI tag

anyway, that's what the latest update uses.

**Comment by ricky-rav** [comment]
That works. The `4.14-4.14` bit seems a bit odd, but no big deal for me. I'm more concerned about why the test keeps failing :-|

**Pull request closed without merging**